embed/etcd.go

```diff
@@ -400,6 +408,29 @@ func (e *Etcd) serve() (err error) {
 			e.errHandler(s.serve(e.Server, &e.cfg.ClientTLSInfo, h, e.errHandler))
 		}(sctx)
 	}
+
+	if len(e.cfg.ListenMetricsUrls) > 0 {
+		// TODO: maybe etcdhttp.MetricsPath or get the path from the user-provided URL
+		metricsMux := http.NewServeMux()
+		metricsMux.Handle("/metrics", prometheus.Handler())
+
+		for _, murl := range e.cfg.ListenMetricsUrls {
+			if murl.Scheme == "https" {
```

Comment:
I misunderstood your comment https://github.com/coreos/etcd/pull/8242#discussion_r127047464

> there may be cases where it's OK to expose the metrics as plaintext but access to the kv store is locked down with tls authentication

I thought we want expose duplicate `/metrics` endpoints with no TLS, when client cert is specified. Or do we just use client cert, then if http://... gets plaintext, https:// gets tls, unix:// gets unix socket, etc.